insights_engine.py

```
@@ -0,0 +1,192 @@
+import pandas as pd
+import calendar
```

Ruff (0.11.9)

`calendar` imported but unused

Remove unused import: `calendar`

(F401)